transformer_engine/pytorch/ops/basic/basic_linear.py

```diff
@@ -936,8 +1062,7 @@ def _functional_backward(
         _wait_async(dy_async)
         _wait_async(x_async)
         _wait_async(dx_async)
-        grad_input = None
-        if dx is not None:
+        if grad_input is None:
```

Comment:
Why removing the check for dx being None?

Reply:
Fixed